### DISCUSSION TITLE
Donation link

### DISCUSSION THREAD

**Discussion by b100dian** [sailfishos-patches/patchmanager#96] [Ideas]
Should we revamp the Donate link just as we did with the About page?
It is still linking to Sfiet's paypal, not even Coderus is mentioned.

I personally don't need the money but I would like to see if people appreciate.
Let me know your thoughts.

There are several other options: for example, I've seen Sebastian from Fernschreiber once saying "If you want to donate, please do so to FSFE". Do you feel there is a similar solution for our case?
(example: pay Web catalog hosting to Basil)

**Comment by Olf0** [comment]
Ack & yes, but not as an example:
I think we should explicitly ask people to donate to Basil for paying the hosting costs for Openrepos via https://openrepos.net/donate

**Reply by Olf0** [reply]
First try writing up something:

If you appreciate our work, please consider a donation for paying the hosting costs for Openrepos via https://openrepos.net/donate

We as well as many others are using this crucial infrastructure for free, both as developers for distributing and as users for downloading software, but tend to forget that is has costs which someone must pay.

**Comment by nephros** [comment]
I also don't require (or even want to accept really) money for the stuff I do on SFOS, including PM.
I like the idea of supporting Basil, +1 from me. Donating to things like the FSFE or e.g. [NOYB](https://noyb.eu) is a good solution as well.  

Removing the Paypal link: Should we contact Sfie for their approval? I would find it a bit rude to rip it out without notification.

And perhaps we can consolidate all the donation options in another Page triggered from the button/link, so we can keep any amount (well, a reasonable amount) of donation links and some explanatory text. So, the Basil link, maybe a organisation donation link and if we want, Sfiets original paypal as well.

**Reply by Olf0** [reply]
> Removing the Paypal link: Should we contact Sfie for their approval? I would find it a bit rude to rip it out without notification.

Try, if you want to be nice, but I would expect no reply (as usual) and he has not touched PM for > 5 years.

**Comment by b100dian** [comment]
1. I think we have a quorum to add Web Catalog hosting as the donation link.
The only problem I know of is that I wasn't able to donate to Openrepos from my country. We should contact Basil maybe he also has paypal, but the card processing on his current link on openrepos did not work for me (even with an Transferwise card which is probably considered UK-issued).

1. I am also OK with multiple donation links / page if it is worth it, and if they are separated like in the about page. But in this case we should _add_ Andrey too. But this would be the last resort option..

1. I like this if 1) doesn't work
> an organisation donation link

If you know how that works (e.g. practically, with a payment system) then you can set it up and it will solve the openrepos donation above too (I mean, it will proxy it, if one of us has it working for openrepos, as I don't)

**Reply by Olf0** [reply]
> 1. I think we have a quorum to add Web Catalog hosting as the donation link.

Yes.
> 2. I am also OK with multiple donation links / page if it is worth it, and if they are separated like in the about page.

Fine fore me, too.
But ...
> But in this case we should _add_ Andrey too.

Not IMO!
The past maintainers had their donations, if they wanted them.
And AFAIR & AFAICS Andrey never wanted any, he simply retained the original PayPal link to donate to Sfietconstatin (Lucien Lu): That is the URL PM 3.0.65 still calls and all PM2 & 3.0 versions provide (likely PM 1, too; I just never used it).
(As a matter of fact Andrey is still an active maintainer by running the sailversion and Web Catalog web-services.)

I would like to get away from personal donations at all, as we do not want them, and Sfietconstatin (Lucien Lu) ceased to maintain PM long ago (and AUSMT is also not used any more).
Side note: @nephros, this is explicitly not about sponsoring Basil (in person), but Openrepos, primarily as the hosting site of the WC (and secondarily as one of our software distribution channels).
> 3. I like this if 1) doesn't work

???  I fail to understand this statement.

> > an organisation donation link

I am also fine with offering a single second option as an alternative (i.e., a secondary, minor option) either the FSFE or NYOB, [which were suggested](https://github.com/sailfishos-patches/patchmanager/discussions/96#discussioncomment-1457950) by @nephros.

But let us avoid a list with many choices: Only a primary option (the one suggested by us) and a secondary as an alternative, if there are technical or political hurdles for the sponsor.

**Reply by nephros** [reply]
> 
> > an organisation donation link
> 
> If you know how that works (e.g. practically, with a payment system) then you can set it up and it will solve the openrepos donation above too (I mean, it will proxy it, if one of us has it working for openrepos, as I don't)

A misunderstanding. I meant "a donation link of an existing organisation" (e.g. FSFE), not our GH organisation here, and not some tool or website.

**Reply by b100dian** [reply]
Yes, brainfart from me, although I mentioned donating to an external organization, I missed that when I saw your comment, Thanks for the clarification.

**Comment by Olf0** [comment]
Some additional thoughts: We should provide two links (primary & alternate) to donate easily, just plain text ("donate to the FSFE") requires some research (Where is FSFE's donation page?), hence it likely lowers the number of donations.

Second try for a small "donation page":

If you appreciate our work, please consider a donation to help covering the hosting costs for Openrepos via https://openrepos.net/donate
Openrepos is critical infrastructure specifically for Patchmanager, because its Web Catalog of patches is hosted there.

If you have technical or other issues with donating to Openrepos, then we appreciate donating to the {FSFE|NYOB} as an alternative: \<link\>

**Reply by Olf0** [reply]
Any opinions WRT FSFE vs. NYOB?
I am undecided; while I know the FSFE well, including its deficits (which are quite some), it has structure and I am less afraid that thing may go fundamentally wrong.  NYOB has a mission statement which is even more important to me, but is quite new, and while I highly value Maximillian Schrems, it is not him running NYOB.
But maybe you both have the same preference and I do not have to decide.

**Reply by nephros** [reply]
As a card-carrying (well, I got them *somewhere*) member of both I can not publicly declare a preference :P :P

One might add the EFF as well, though they are very American (for good reason too).

So how about:

> If you appreciate our work, please consider a donation to help covering the hosting costs for Openrepos via https://openrepos.net/donate
Openrepos is critical infrastructure specifically for Patchmanager, because its Web Catalog of patches is hosted there.
>
 > If for some reason you can not donate to Openrepos, why not support one of the following organisations: 
>  - FSFE \<web page\> \<donation link\>
>  - NYOB \<web page\> \<donation link\>
> - ...

... and each of us gets to nominate exactly one organisation. That's three alternatives, perfect.

**Reply by Olf0** [reply]
???
You nominating two organizations, Vlad and me none, does not make "each of us gets to nominate exactly one organisation". 

I also do not think "that's three alternatives, perfect.", because we sure could name 10 organisations which are somehow O.K.: I actually want to guide people to donate for OpenRepos, and do not want this message to be diluted by a lot of alternatives.
That is why I tried to determine which (single) alternative ranks highest among us (because of the technical issues donating to OpenRepos might pose).
Just to start with naming a preference, I now decided for the FSFE.

**Reply by b100dian** [reply]
FSFE appeared in messages from me too, so let's go with this for the alternative.
For openrepos, remember that we are also depending on Andrey to deploy Web catalog if we need changes. Also there's a database of patches there. I _think_ we should fix this in the long run by taking that over too, but my django coding was only monkey-see mokey-do so I cannot say for sure I'm up to do it (though on the next occasion - bitness support was pretty close - I will). 
But then again Andrey was ok without a donation link in PM itself for so long.

So:
- Openrepos
and alternatively
- FSFE
it is?

There's only one thing that "bothers" me, is that we don't have a measure for this as a 'success rate'. We can try to change the wording so that people would get to write 'for patchmanager' or something, but no pingback will happen anyway. Oh well, I can leave without that.

**Reply by Olf0** [reply]
#### v0.3 

If you appreciate our work, please consider a donation to help covering the hosting costs for Openrepos via https://openrepos.net/donate
Openrepos is critical infrastructure specifically for Patchmanager, because its Web Catalog of patches is hosted there.

If for some reason you can not donate to Openrepos, we also appreciate donating to the Free Software Foundation Europe (FSFE): https://fsfe.org/donate

**Reply by nephros** [reply]
> #### v0.3
> If you appreciate our work, please consider a donation to help covering the hosting costs for Openrepos via https://openrepos.net/donate Openrepos is critical infrastructure specifically for Patchmanager, because its Web Catalog of patches is hosted there.
> 
> If for some reason you can not donate to Openrepos, we also appreciate donating to the Free Software Foundation Europe (FSFE): https://fsfe.org/donate

Done! :)
https://github.com/sailfishos-patches/patchmanager/pull/110

**Comment by Olf0** [comment]
Deleting this discussion thread, because we are done.

**Reply by Olf0** [reply]
Or shall we keep it for documentation purposes?
Then I would prefer to move this discussion to an issue and close it.

Ah, we have to try procedures like this in order to determine what is best suited.
So, let's go!

**Reply by Olf0** [reply]
Good that I did experiment a bit:

"Create issue from discussion" just copies the initial post to a new issue, but not the discussion thread!  So that is not suited to archive this discussion.

"Transfer discussion" provides the ability to move this discussion to another repository.  Nothing which is really useful for us, IMO.

So all one can do for archiving a discussion is to "lock" it.  We as project maintainers can still comment then.  And we can "unlock" it again for everybody at any time.

Thus a "locked" discussion is rather an indicator the it was finished, than a real measure.
